### PR TITLE
4 patch articles/article

### PR DESCRIPTION
**Pull request by Cralus**
implemented patch articles by article id with associated testing and error handling for bad request body

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -6,81 +6,139 @@ const app = require("../app.js");
...
+  test('Should respond with a 400 error if given a request body with incorrect formatting of inc_votes', () => {
+    return request(app)
+    .patch("/api/articles/1")
+    .expect(400)
+    .send({ wrongName: 10 })
+    .then(({ body: { msg } }) => {
+         expect(msg).toBe('Bad Request')
+    })
+  });
+  test('should respond with a 400 error if given an object with too many properties', () => {
```
Comment (cjplatten) [inline __tests__/app.test.js]: This test description doesn't match the test performed, what are you testing for here?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -6,81 +6,139 @@ const app = require("../app.js");
...
+    return request(app)
+    .patch("/api/articles/1")
+    .expect(400)
+    .send({ wrongName: 'hello' })
+    .then(({ body: { msg } }) => {
+         expect(msg).toBe('Bad Request')
     })
   });
-  test('should return a status of 400 if not given a number and a msg of bad request', () => {
+  test('should respond with a 400 error if given an object with the inc_votes property but with the incorrect data type', () => {
```
Comment (cjplatten) [inline __tests__/app.test.js]: same comment as above

File: __tests__/app.test.js [hunk 3]
```diff
@@ -6,81 +6,139 @@ const app = require("../app.js");
...
+    .patch("/api/articles/1")
     .expect(400)
-    .then(({body: {msg}}) => {
-        expect(msg).toBe('Bad Request')
+    .send({ wrongName: 10, shouldntbehere :10 })
+    .then(({ body: { msg } }) => {
+         expect(msg).toBe('Bad Request')
     })
   });
 });
```
Comment (cjplatten) [inline __tests__/app.test.js:152]: How would you want your server to respond if there were issues with the article id in the path or if no article could be found?

File: models/articles.js [hunk 4]
```diff
@@ -1,14 +1,29 @@
...
+}
+exports.updateArticlesById = (article, votes) => {
+    article.votes += votes
+    
+    return db.query(
+        `UPDATE articles SET votes = $1 WHERE article_id = $2;`, [article.votes, article.article_id]
+    ).then(() => {    
+    return db
+    .query(
+        `SELECT * FROM articles WHERE article_id = $1;`, [article.article_id]
```
Comment (cjplatten) [inline models/articles.js]: While this works fine you don't have to make this second query in order to get the updated article back from the database, have a think about how you would get this data from your initial query.

**Review by cjplatten — CHANGES_REQUESTED** [review]
Have a look at the comments and please submit another PR once you've addressed them

**Comment by Cralus** [comment]
Implemented all the comments and suggestions for refactor

**Review by cjplatten — APPROVED** [review]
Looks good!

**Pull request merged**